src/scene/light.cpp

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2021 F. Lotfi & D. Kane All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+#include "light.h"
+
+Light::Light(glm::vec3 direction) : mDirection(direction) {}
+float Light::Intensity(const Triangle &triangle) const {
+  return glm::dot(triangle.normal(), mDirection);
+}
+
+glm::vec3 Light::GetColor(const Triangle &triangle) const {
+  float intensity = Intensity(triangle);
+  return triangle.color() * intensity;
```

Comment:
Lights can have their own color (not just white light). Can we have a vec3 that contains that information for each light. I'm also not sure why the color of the triangle is relevant in color's getColor function. I know that you will have to use triangle's color in the lighting equation eventually but I don't think it should be in the light class.

Reply:
You bring up a great point about lights having color. The GetColor api is  to eventually account for that. What we have now is essentially two types of color. the color of the surface and the color of the light. for now b\c there is a texture PR in the works I'm going to address this in a separate PR.

For your second point on color in the light class. Light doesn't store any color in this API, it just computes based on the direction of light. In fact we only change the color in the hit class, the color of the triangle surface remains unchanged.

Other than the hue of the light which I mentioned would be address above, there is no other plan to store color in the light api.